internal/component/database_observability/mysql/collector/query_sample.go

```diff
@@ -184,6 +192,12 @@ func (c *QuerySample) initializeBookmark(ctx context.Context) error {
 }
 
 func (c *QuerySample) fetchQuerySamples(ctx context.Context) error {
+	if c.autoEnableSetupConsumers {
+		if err := c.updateSetupConsumersSettings(ctx); err != nil {
+			return err
+		}
+	}
+
```

Comment:
As commented by @gaantunes we could decouple the update from the regular collector loop. I'm keen to postpone that work for a followup PR, here I'm more interested in the correctness of the approach.